sendmail/sendmail.mli

```diff
@@ -16,9 +16,9 @@ type authentication =
 type ('a, 's) stream = unit -> ('a option, 's) io
 
 type error =
-  [ Request.Decoder.error
-  | Reply.Decoder.error
-  | `Unexpected_response of int * string list
+  [ `Error of [ Request.Encoder.error
```

Comment:
This is a [canonic example](https://paste.x25519.net/woI?hl=ocaml#mMeWNdZKPX7exuyoT+9HR86iZ11x2hCY3CMht1wBk2o=) of what I really want. The point is to be able to let the user to define some errors. About that, the type of functions (to be able to extend the error type) should be `val foo : ... -> (_, [> error ]) result`.

But the type error must be a polymorphic variant. The only way (from what I know) to compose an error type from the given module to larger error type (defined by the functor) and keep that the final type __is__ a polymorphic variant is to wrap the error type of the given module to a new constructor `Error` (which, I know, is not the best name).

Reply:
I see the point of the PR but I don't see why this specific change was needed. The argument of `` `Error `` is a polymorphic variant too (but closed).
Is it to avoid future collisions with user errors ?

Reply:
> Is it to avoid future collisions with user errors ?

Exactly, again, it's not the best and a proper solution but the trade-off still is good when this update is only about errors. At the end, this update is needed for [ptt](https://github.com/dinosaure/ptt) which go further about SMTP protocol.